src/main/java/com/mateacademy/service/UserServiceImp.java

```diff
@@ -0,0 +1,43 @@
+package com.mateacademy.service;
+
+import com.mateacademy.model.UserModel;
+import com.mateacademy.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class UserServiceImp implements UserService {
```

Comment:
Please add tests for this class. 
https://stackoverflow.com/questions/2288386/mockito-how-to-test-my-service-with-mocking